aleph/views/mappings_api.py

```diff
@@ -0,0 +1,123 @@
+import logging
+
+from flask import Blueprint, request
+# from followthemoney import model
+# from followthemoney.exc import InvalidMapping
+from werkzeug.exceptions import BadRequest
+
+from aleph.core import db, archive
+from aleph.model import Mapping
+from aleph.search import QueryParser, DatabaseQueryResult
+from aleph.views.serializers import MappingSerializer, first
+from aleph.views.util import get_db_collection, get_index_entity, parse_request  # noqa
+from aleph.views.context import enable_cache
+from aleph.views.forms import MappingSchema
+from aleph.views.util import require, obj_or_404, get_session_id
+from aleph.queues import queue_task, OP_BULKLOAD
+from aleph.logic.collections import refresh_collection
+
+blueprint = Blueprint('mappings_api', __name__)
+log = logging.getLogger(__name__)
+
+
+@blueprint.route('/api/2/collections/<int:collection_id>/mappings', methods=['GET'])  # noqa
+def index_by_collection(collection_id):
+    collection = get_db_collection(collection_id)
+    parser = QueryParser(request.args, request.authz)
+    q = Mapping.by_collection(collection.id)
+    if 'table' in parser.filters:
+        q = Mapping.by_table(parser.filters['table'], q)
+    result = DatabaseQueryResult(request, q, parser=parser)
+    return MappingSerializer.jsonify_result(result)
+
+
+def _load_query():
+    try:
+        query = request.json.get('mapping_query', '{}')
+        # TODO: validate query
```

Comment:
I wonder if we should do a refactor in `followthemoney` that makes it validate both mappings and schemata against a written-out JSONSchema spec. At the moment, the validation is a little ad-hoc and procedural, and if we have JSONSchema files for these formats they might also come in handy with the swagger docs. WDYT?